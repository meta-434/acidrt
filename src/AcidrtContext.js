import React from "react";

const AcidrtContext = React.createContext({});

export default AcidrtContext;
