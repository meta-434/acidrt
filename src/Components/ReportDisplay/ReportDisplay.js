import React, { Component } from 'react';
import ErrorDisplay from "../ErrorDisplay/ErrorDisplay";
import AcidrtContext from "../../AcidrtContext";
import Form from "../Form/Form";

export default class ReportDisplay extends Component {

    static contextType = AcidrtContext

    state = {
        uniqueReport: undefined,
        error: undefined,
        editMode: false,
        oldReports: undefined,
    }

    componentDidMount() {
        this.findUniqueReport();
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
        if (this.context.reports !== this.state.oldReports) {
            this.findUniqueReport()
        }
    }

    findUniqueReport = () => {
        const srcId = parseInt(this.props.match.params.reportId, 10);
        const srcResult = this.context.reports.filter(report => report.id === srcId)
        if (srcResult.length === 1) this.setState({uniqueReport: srcResult[0], oldReports: this.context.reports})
    }

    handleDelete = (reportId) => {
        this.context.handleDeleteReport(reportId);
        this.props.history.push('/all-reports');
    }

    toggleEdit = () => {
        const isEditMode = this.state.editMode;
        this.setState({editMode: !isEditMode});
    }

    render(){
        const rep = this.state.uniqueReport;
        if (!!this.state.uniqueReport) {
            if (!this.state.editMode) {
                return (
                    <div>
                        {!!this.context.error ? <ErrorDisplay /> : ""}
                        <section>
                            <h2>Report Details</h2>
                        </section>
                        <section>
                            <p>id: {rep.id}</p>
                            <p>Report for incident at lat: {rep.report_lat} lng: {rep.report_lng}</p>
                            <p>Report Submitted By: {rep.report_first + " " + rep.report_last}</p>
                            <p>incident occurred on {rep.report_date} at {rep.report_time}</p>
                            <div id={'incident-map'}>
                                Incident Type(s):
                                {
                                    (rep.report_type.length > 1) ? (<p>{rep.report_type}</p>) : (<ul>{rep.report_type.map((el, idx) => <li key={idx}>{el}</li>)}</ul>)
                                }

                            </div>

                            <p>if other, describe type: {(rep.report_other !== 'undefined') ? (rep.report_other) : ('no -other- details given')}</p>
                            <p>{(rep.report_waterbody) ? ('Body of water affected: ' + rep.report_waterbody) : ('no body of water given')}</p>
                            <p>Submitter Contact info: {rep.report_email}, {rep.report_phone}</p>
                            <p>extra details, if any: {(rep.report_details !== "undefined") ? (rep.report_details) : ('no extra details given')}</p>
                            <button onClick={() => this.handleDelete(rep.id)}>Resolve (delete)</button>
                            <button onClick={this.toggleEdit}>Edit</button>
                        </section>
                    </div>
                );
            } else {
                return (
                    <div>
                        <p>edit mode!!</p>
                        <Form defaultVals={this.state.uniqueReport} editMode={this.state.editMode} toggleEdit={() => this.toggleEdit()}/>
                    </div>

                );
            }

        } else {
            return (
                <p>loading...</p>
            )
        }
    }
}