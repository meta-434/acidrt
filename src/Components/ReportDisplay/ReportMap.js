import React, { Component } from 'react';
import {Map, InfoWindow, Marker, GoogleApiWrapper} from 'google-maps-react';
import AcidrtContext from "../../AcidrtContext";

export class MapComponent extends Component {

    static contextType = AcidrtContext;

    state={

    }


    render() {
        console.log('maaaa', this.props);
        const containerStyle = {position:'relative', display: 'block', width: '50vh', height:'50vh'}
        const { markers } = this.props;
        return(
            <Map
                containerStyle={containerStyle}
                style={{width: '100%', height: '100%', display: 'block', position: 'relative', overflowY: 'hidden'}}
                google={this.props.google}
                initialCenter={ { lat: 38.028, lng: -78.5635 } }
                zoom = { 9 }
            >
                {
                    markers.map((marker, idx) => {
                        console.log('marker at', marker)
                        return (
                            <Marker
                                key={idx}
                                name={`${marker.info}`}
                                position={{ lat: `${marker.lat}`, lng: `${marker.lng}` }}
                            />
                        )
                    })
                }
            </Map>
        );
    }
};
export default GoogleApiWrapper({
    apiKey: (process.env.REACT_APP_MAPS_KEY),
})(MapComponent);