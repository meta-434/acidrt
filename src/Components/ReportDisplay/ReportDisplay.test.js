import React from 'react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom';
import AcidrtContext from "../../AcidrtContext";
import {BrowserRouter, Route} from "react-router-dom";
import ReportDisplay from "./ReportDisplay";

test('NotesDisplay renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <AcidrtContext.Provider value={{
                reports: [
                    {
                        id: 0,
                        first: 'john',
                        last: 'doe',
                        email: 'email@email.com',
                        lat: 38.030425,
                        lng: -78.561065,
                        error: undefined,
                        date: '2020-02-02',
                        time: '12:01',
                        phone: '123-123-1234',
                        type: ["Suspicious discharge from pipe into stream"],
                        waterBody: 'James River'
                    },
                    {
                        id: 1,
                        date: "2020-01-01",
                        details: "take 4 lefts.",
                        email: "acyrush@gmail.com",
                        error: undefined,
                        first: "Alexandre",
                        last: "Hapgood",
                        lat: 38.09802480089654,
                        lng: -78.4887256216315,
                        other: "lorem ipsum other details",
                        phone: "434-249-7488",
                        time: "12:34",
                        type: ["Suspicious discharge from pipe into stream", "Suspicious suds or other substance floating on water"],
                        waterBody: "oh so much water in the lake",
                    }
                ],
            }}>
                <Route path="/all-reports/:reportId" render={props => <ReportDisplay {...props} />} />
            </AcidrtContext.Provider>
        </BrowserRouter>
        , div);
    ReactDOM.unmountComponentAtNode(div);
})

test('ReportDisplay renders correctly', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <AcidrtContext.Provider value={{
                    reports: [
                        {
                            id: 0,
                            first: 'john',
                            last: 'doe',
                            email: 'email@email.com',
                            lat: 38.030425,
                            lng: -78.561065,
                            error: undefined,
                            date: '2020-02-02',
                            time: '12:01',
                            phone: '123-123-1234',
                            type: ["Suspicious discharge from pipe into stream"],
                            waterBody: 'James River'
                        },
                        {
                            id: 1,
                            date: "2020-01-01",
                            details: "take 4 lefts.",
                            email: "acyrush@gmail.com",
                            error: undefined,
                            first: "Alexandre",
                            last: "Hapgood",
                            lat: 38.09802480089654,
                            lng: -78.4887256216315,
                            other: "lorem ipsum other details",
                            phone: "434-249-7488",
                            time: "12:34",
                            type: ["Suspicious discharge from pipe into stream", "Suspicious suds or other substance floating on water"],
                            waterBody: "oh so much water in the lake",
                        }
                    ]
                }}>
                    <Route path="/all-reports/:reportId" render={props => <ReportDisplay {...props} />} />
                </AcidrtContext.Provider>
            </BrowserRouter>
        )
        .toJSON();

    // console.log(tree);
    expect(tree).toMatchSnapshot();
});
