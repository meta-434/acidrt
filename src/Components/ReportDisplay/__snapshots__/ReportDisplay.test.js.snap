// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDisplay renders correctly 1`] = `null`;
